TelepresenceBot/core/src/main/java/com/novoda/tpbot/human/HumanPresenter.java

```diff
@@ -42,6 +47,7 @@ public void update(Result result) {
             if (result.isError()) {
                 humanView.onError(result.exception().get().getMessage());
             } else {
+                lastServerPersistence.saveLastConnectedServer(serverAddress);
```

Comment:
this is fine 👍 , but you know a presenter is the "glue" between the model and the view.

you could argue that what you have here is still just glueing the two together, (the model being `HumanTelepresenceService` & `LastServerPersistence`) as it gets more complicated your presenter gets more and more collaborators.

If your presenter had only one dependency something like `HumanModel` (and this was composed of `HumanTelepresenceService` & `LastServerPersistence`) then the presenter stays relatively simple "gluing" the model to the view.
It will be your model that expands and gets bigger and bigger (which is awesome because this is where your domain is.)  Then if you decided "ok now we save the last connected address when the connection is established but I actually want to save it when the connection is attempted" the change would be made in the model (currently the change would be made in the presenter), this is clearly a domain & model decision - not the presenters, who should just be presenting data for the UI to be shown.

Food for thought :-)